flask_monitoringdashboard/core/measurement.py

```diff
@@ -104,29 +106,40 @@ def evaluate(route_handler, args, kwargs):
     :param kwargs:
     :return:
     """
-    try:
-        result = route_handler(*args, **kwargs)
-        status_code = status_code_from_response(result)
+    g.scoped_logger = ScopedExceptionLogger()
+
+    def evaluate_():
+        try:
+            result = route_handler(*args, **kwargs)
+            status_code = status_code_from_response(result)
+
+            return result, status_code
+        except BaseException as e:
+            g.scoped_logger.uncaught_exception_info = sys.exc_info()
+            if isinstance(e, HTTPException):
+                return None, e.code
+            return None, 500
+
+    result, status_code = evaluate_()
+    if len(g.scoped_logger.user_captured_exceptions) != 0 or g.scoped_logger.uncaught_exception_info is not None:
```

Comment:
Can we add a little comment on how this exception handling code works at the top of the evaluate method? I can't figure out why `g.scoped_logger.user_captured_exceptions` would ever be non-empty, given that the only thing that we do in evaluate is to assign `uncaught_exception_info`.